pkg/controller/volume/scheduling/scheduler_binder.go

```diff
@@ -485,6 +513,11 @@ func (b *volumeBinder) checkBindings(pod *v1.Pod, bindings []*bindingInfo, claim
 			return false, nil
 		}
 
+		pv, err = b.tryTranslatePVToCSI(pv, csiNode)
+		if err != nil {
+			return false, fmt.Errorf("failed to translate pv to csi: %v", err)
+		}
+
 		// Check PV's node affinity (the node might not have the proper label)
 		if err := volumeutil.CheckNodeAffinity(pv, node.Labels); err != nil {
```

Comment:
Since we're not doing the translation inside CheckNodeAffinity, then that means we also need to add something to the kubelet path, which also validates node affinity.